e2e/multi-pipeline/pipeline_test.go

```diff
@@ -0,0 +1,525 @@
+			})
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("prod env init should succeed", func() {
+			_, err := copilot.EnvInit(&client.EnvInitRequest{
+				AppName: appName,
+				EnvName: "prod",
+				Profile: "e2eprodenv",
+			})
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("env ls should list both envs", func() {
+			out, err := copilot.EnvList(appName)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(out.Envs)).To(Equal(2))
+			envs := map[string]client.EnvDescription{}
+			for _, env := range out.Envs {
+				envs[env.Name] = env
+				Expect(env.ExecutionRole).NotTo(BeEmpty())
+				Expect(env.ManagerRole).NotTo(BeEmpty())
+			}
+
+			Expect(envs["test"]).NotTo(BeNil())
+			Expect(envs["prod"]).NotTo(BeNil())
+			Expect(envs["test"].Region).NotTo(Equal(envs["prod"].Region))
+			Expect(envs["test"].Account).NotTo(Equal(envs["prod"].Account))
+		})
+	})
+
+	Context("when creating the frontend service", func() {
+		It("should initialize the service", func() {
+			_, err := copilot.SvcInit(&client.SvcInitRequest{
+				Name:       "frontend",
+				SvcType:    "Load Balanced Web Service",
+				Dockerfile: "./frontend/Dockerfile",
+				SvcPort:    "80",
+			})
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("should generate a manifest file", func() {
+			Expect(filepath.Join(repoName, "copilot", "frontend", "manifest.yml")).Should(BeAnExistingFile())
+		})
+		It("copies a template to an addons file", func() {
+			cmd := exec.Command("cp", "s3template.yml", filepath.Join(repoName, "copilot", "frontend", "addons", "e2e-pipeline-addon.yml"))
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			Expect(cmd.Run()).NotTo(HaveOccurred())
+		})
+		It("should list the service", func() {
+			out, err := copilot.SvcList(appName)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(out.Services)).To(Equal(1))
+			Expect(out.Services[0].Name).To(Equal("frontend"))
+		})
+	})
+
+	Context("when creating the test pipeline manifest", func() {
+		It("should initialize the pipeline", func() {
+			_, err := copilot.PipelineInit(client.PipelineInitInput{
+				Name:         testPipelineName,
+				URL:          repoURL,
+				GitBranch:    "test",
+				Environments: []string{"test"},
+			})
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("should generate pipeline artifacts", func() {
+			Expect(filepath.Join(repoName, "copilot", "pipelines", testPipelineName, "manifest.yml")).Should(BeAnExistingFile())
+			Expect(filepath.Join(repoName, "copilot", "pipelines", testPipelineName, "buildspec.yml")).Should(BeAnExistingFile())
+		})
+		It("should push repo changes upstream", func() {
+			cmd := exec.Command("git", "config", "user.email", "e2etest@amazon.com")
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			cmd.Dir = repoName
+			Expect(cmd.Run()).NotTo(HaveOccurred())
+
+			cmd = exec.Command("git", "config", "user.name", "e2etest")
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			cmd.Dir = repoName
+			Expect(cmd.Run()).NotTo(HaveOccurred())
```

Comment:
nit: can we move these to the `"setup CodeCommit repository"` section so that we need to repeat ourselves while pushing to the separate branches